app/src/main/java/org/schabi/newpipe/util/StreamDialogEntry.java

```diff
@@ -206,4 +229,30 @@ private static void openChannelFragment(final Fragment fragment,
                 fragment.requireActivity().getSupportFragmentManager(),
                 item.getServiceId(), uploaderUrl, item.getUploaderName());
     }
+
+    private static boolean isNetworkAvailable(final Context context) {
+        final ConnectivityManager cm =
+                (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+
+        final NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
+        return activeNetwork != null
+                && activeNetwork.isConnectedOrConnecting();
+    }
+
+
+    /////////////////////////////////////////////////////////
+    // private method to show unavailable network Snackbar //
+    /////////////////////////////////////////////////////////
+
+    private static void showNetworkUnavailableSnackbar(final Fragment fragment) {
+        final Snackbar snackbar = Snackbar.make(
+                fragment.requireView(),
+                R.string.error_no_network,
+                Snackbar.LENGTH_INDEFINITE);
```

Comment:
Uhm, indefinite length is not a good idea, I'd just use `LENGTH_SHORT`, it doesn't take much time to read a snackbar that says "Network not available"

Reply:
How short is `LENGTH_SHORT`? 5 seconds?

Reply:
2 seconds (LENGTH_LONG is 3.5 seconds): https://stackoverflow.com/questions/7965135/what-is-the-duration-of-a-toast-length-long-and-length-short